tests/integration/index/drop_test.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package index
+
+import (
+	"testing"
+
+	testUtils "github.com/sourcenetwork/defradb/tests/integration"
+)
+
+func TestIndexDrop_ShouldNotHinderQuerying(t *testing.T) {
```

Comment:
todo: I think it well worth also adding a test for dropping an index that does not exist.